macros/utils/aggregations/_min_or_max.sql

```diff
@@ -1,56 +1,65 @@
 {% macro _min_or_max(min_or_max, qualified_col) %}
 
+{% if target.type == "snowflake" %}
```

Comment:
Great work! But I think we ought to change this to use [dbt dispatch](https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch#examples). 

I think we could just abstract this by doing `{% set aggregation = get_db_aggregation() %}`. And have that `get_db_aggregation()` macro contain `default__` and `snowflake__` dispatches.

Sound reasonable?

Reply:
Yep, absolutely! Wasn't aware of that functionality but seems like a perfect fit here. Will make the changes & tee this up!

Reply:
Hi @tnightengale:

- Restated the file without `sqlfmt` so the diff should be cleaner.
- Struggling a bit with how the `dbt dispatch` logic. I see how it can be used to replace the if/then logic I was using and have done so, but I'm not following applying this at the `{% set aggregation = get_db_aggregation() %}` level since there are other changes to the Snowflake specific logic. Would you be able to spell out a bit further what you are looking for there? Happy to take another pass but recognize that I'm spinning my wheels.

# Validation

1) No row differences between dev and prod pre change
![image](https://user-images.githubusercontent.com/15633266/233718030-9476d685-8e68-47a1-94ca-bee657c8e748.png)
2) No row differences between dev and prod post change
![image](https://user-images.githubusercontent.com/15633266/233719971-1515e4fc-a159-443f-8eea-843a6affa31d.png)